version: '3.8'

services:
  postgres:
    container_name: postgres-rest
    image: postgres:15.1-alpine
    restart: on-failure
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    env_file:
      - .env
    networks:
      - postgresnet

  redis:
    image: redis:latest
    container_name: redis
    restart: always
    volumes:
      - redis_data:/data
    expose:
      - "6379"
    ports:
      - "6379:6379"
    networks:
      - webnet
    env_file:
      - .env
    healthcheck:
      test: ["CMD-SHELL", "redis-cli ping | grep PONG" ]
      interval: 1s
      timeout: 3s
      retries: 5

  ray_api:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: ray_api
    depends_on:
       redis:
         condition: service_healthy
    env_file:
      - .env
    volumes:
      - .:/app
    restart: always
    ports:
      - "8000:8000"
    networks:
      - postgresnet
      - webnet
    command: /start

  celery:
    build: .
    command: celery -A ray_api worker -l info -P solo
    volumes:
      - ./backups/:/app/backups
    depends_on:
      - ray_api
      - redis
    env_file:
      - .env
    networks:
      - postgresnet
      - webnet
  celery_beat:
    build: .
    volumes:
      - celery_beat:/data
    command: celery -A ray_api beat
    depends_on:
      - ray_api
      - redis
    env_file:
      - .env
    networks:
      - postgresnet
      - webnet

networks:
  postgresnet:
  webnet:

volumes:
  postgres_data:
  redis_data:
  celery_beat:
  static_value:
